core/templates/pages/exploration-editor-page/services/router.service.ts

```diff
@@ -165,11 +165,21 @@ angular.module('oppia').factory('RouterService', [
     };
 
     var _savePendingChanges = function() {
-      // Sometimes, AngularJS throws a "Cannot read property $$nextSibling of
-      // null" error. To get around this we must use $timeout().
-      $timeout(function() {
+      try {
         $rootScope.$broadcast('externalSave');
-      });
+        // Important: The log call below is for only unit tests purposes.
```

Comment:
In general, never add code that is only for unit test purposes. I think it's necessary to figure out why catch() isn't called when $broadcast leads to errors. Otherwise what may happen is that, in production, $rootScope.$broadcast errors but the catch() is still not called (and yet the tests would pass).

Reply:
The catch is being called but for some reason, it still fails the test. My guess is that when mocking errors on AngularJS native methods, the code works as expected but jasmine can't handle it and fails.
When I mocked $broadcast, the function was being fully covered but it wasn't passing because of the jasmine fail.

Reply:
Do you have evidence to support your guess (like an external link that says "this is a known issue when mocking errors on Jasmine")? It's quite important to get that evidence, otherwise you might be applying the wrong fix and hiding some other issue.

One other thought is whether the bullet points in https://github.com/oppia/oppia/wiki/Frontend-test-best-practices#best-practices-for-good-tests are relevant -- e.g. the use of done() to mark the test as asynchronous so that it waits for all the async calls to finish. Also take care to make sure that your expectation is only being checked after the async calls is made, otherwise you could get the wrong result because you are checking the expectation before it has time to be satisfied.

Reply:
I don't think `done` works here, because it's not an asynchronous call. I've done some search using old versions and what I've been trying to do worked here: https://jsfiddle.net/3mnbLp5x/3/

My guess is wrong now I just need to find why this is happening in newest versions. But for now using another service is the only thing that worked.

Reply:
OK, thanks! Let me know once you have a theory that works.

Reply:
I've found a solution, PTAL!